### PR TITLE
chore(design): Added descriptions to semantic tokens JSON files

### PR DESCRIPTION
**Pull request by edison-cy-yang**
## Motivations

This is an enabler so we can use semantic descriptions in docs

### Added

- 

### Changed

- 

### Deprecated

- 

### Removed

- 

### Fixed

- 

### Security

- 

## Testing

Changes can be
[tested via Pre-release](https://github.com/GetJobber/atlantis/actions/workflows/trigger-qa-build.yml)

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

Random photo of Atlantis

### CODE REVIEW

**Comment by edison-cy-yang** [comment]
@chris-at-jobber A couple of questions that I came across while adding these descriptions:

- should some of the tokens json structures be changed, eg dont make `subtle` on the same level as `hover` in `interactive-` of `semanticColor.tokens.json`?

- semantic colors for workflows in light mode are set in `workflow.tokens.json` but the ones for dark mode are set in `dark.tokens.json`. Should we make the locations of these tokens consistent?

- Should the same description be applied to the hover, surface, and onSurface states of a semantic color?

**Comment by chris-at-jobber** [comment]
@edison-cy-yang 

### Token structure
If that would change the output values then I would say no, as that would be a massive breaking change, but if it's not, then maybe? I'm not sure what exactly that change would look like.

### Workflow colors
I would be OK with moving the token definitions from `workflow.tokens` into `semanticColor.tokens`, I'm not sure if that would have any knock-on effects (@scotttjob might know!)

### Description for hover/etc 
Couldn't hurt! I think as a general rule for 
- hover tokens
  - "Hover state for {the token it's referencing}"
- surface/onSurface tokens
  - "Use for the surface of an element that has {status it's referencing} status"
  - "Use for text or iconography that sits overtop a {same status surface value} background"

**Comment by scotttjob** [comment]
> If that would change the output values then I would say no, as that would be a massive breaking change, but if it's not, then maybe? I'm not sure what exactly that change would look like.

Changing the structure of the JSON would definitely change the final token value. (each 'level' traversed in the JSON file is translated to a dash in the final token)

> I would be OK with moving the token definitions from `workflow.tokens` into `semanticColor.tokens`, I'm not sure if that would have any knock-on effects (@scotttjob might know!)

The only concern I have is for the token aliases, I believe they'll still be processed properly but would want to verify that.

Ideally the only thing that would change from this would be potentially the tokens appearing in a different order in the generated output, but the contents would be identical. So I would be comfortable with this update if I could see that Diff as part of the PR.

**Comment by edison-cy-yang** [comment]
@chris-at-jobber @scotttjob Thanks! I will leave those structural JSON changes out for now. I want to take a look at parsing token descriptions next, hoping I will have more understanding and feel more comfortable making these by then.

Descriptions for hover, surface, and onSurface states added.

File: packages/design/src/tokens/dark.tokens.json [hunk 1]
```diff
@@ -19,248 +19,313 @@
...
             "onSurface": {
-                "$value": "{color.base.blue-.200}"
+                "$value": "{color.base.blue-.200}",
+                "$description": "Use for text or iconography that sits overtop an inactive background."
             }
         },
         "heading": {
-            "$value": "{color.base.blue-.100}"
+            "$value": "{color.base.blue-.100}",
+            "$description": "Headings have a bold, dark color to cement their hierarchy."
```
Comment (chris-at-jobber) [inline packages/design/src/tokens/dark.tokens.json:133]:
```suggestion
            "$description": "Headings have high contrast against the application surface to cement their hierarchy."
```

This is an oversight on my part on the content side, just going to make a few of these suggestions where we're referencing "dark" or "light" directly in the semantic tokens to be more _semantic_ and less _descriptive_. (I'll make these changes in the Figma as well!)

File: packages/design/src/tokens/dark.tokens.json [hunk 2]
```diff
@@ -19,248 +19,313 @@
...
         "text-": {
             "secondary": {
-                "$value": "{color.base.blue-.400}"
+                "$value": "{color.base.blue-.400}",
+                "$description": "Text that is relevant but less important can be lower-contrast to suggest its’ reduced importance."
             },
             "reverse": {
-                "$value": "{color.base.blue-.700}"
+                "$value": "{color.base.blue-.700}",
+                "$description": "Use on “reversed” (light) sections."
```
Comment (chris-at-jobber) [inline packages/design/src/tokens/dark.tokens.json:146]:
```suggestion
                "$description": "Use for typographic content against reversed (inverted theme) surfaces."
```

File: packages/design/src/tokens/dark.tokens.json [hunk 3]
```diff
@@ -19,248 +19,313 @@
...
             },
             "reverse-": {
                 "secondary": {
-                    "$value": "{color.base.blue-.900}"
+                    "$value": "{color.base.blue-.900}",
+                    "$description": "Secondary text for “reversed” sections."
```
Comment (chris-at-jobber) [inline packages/design/src/tokens/dark.tokens.json:151]:
```suggestion
                    "$description": "Secondary text color for use against reversed (inverted theme) surfaces."
```

File: packages/design/src/tokens/dark.tokens.json [hunk 4]
```diff
@@ -19,248 +19,313 @@
...
                 }
             }
         },
         "icon": {
             "$value": "{color.base.blue-.200}"
         },
         "surface": {
-            "$value": "{color.base.blue-.900}"
+            "$value": "{color.base.blue-.900}",
+            "$description": "Most elements in Jobber have a white surface; if clickable, they have a light hover and active color."
```
Comment (chris-at-jobber) [inline packages/design/src/tokens/dark.tokens.json:160]:
```suggestion
            "$description": "The primary color of the application surface (background). If interactive, surfaces have a hover and active color."
```

File: packages/design/src/tokens/dark.tokens.json [hunk 5]
```diff
@@ -19,248 +19,313 @@
...
                 "subtle": {
-                    "$value": "{color.base.blue-.800}"
+                    "$value": "{color.base.blue-.800}",
+                    "$description": "Use when a subtler distinction than the standard surface--background is needed."
                 }
             },
             "reverse": {
-                "$value": "{color.base.taupe-.200}"
+                "$value": "{color.base.taupe-.200}",
+                "$description": "When a strong constrast is needed with the rest of the application, a reversed surface can be used."
```
Comment (chris-at-jobber) [inline packages/design/src/tokens/dark.tokens.json:187]:
```suggestion
                "$description": "Use a reverse surface to invert the theme when a strong contrast is needed with the rest of the application"
```

File: packages/design/src/tokens/semanticColor.tokens.json [hunk 6]
```diff
@@ -3,160 +3,201 @@
...
       "onSurface": {
-        "$value": "{color.base.blue-.900}"
+        "$value": "{color.base.blue-.900}",
+        "$description": "Use for text or iconography that sits overtop an inactive background."
       }
     },
     "heading": {
-      "$value": "{color.base.blue-.900}"
+      "$value": "{color.base.blue-.900}",
+      "$description": "Headings have a bold, dark color to cement their hierarchy."
```
Comment (chris-at-jobber) [inline packages/design/src/tokens/semanticColor.tokens.json:121]:
```suggestion
      "$description": "Headings have a high contrast against the application surface to cement their hierarchy."
```

File: packages/design/src/tokens/semanticColor.tokens.json [hunk 7]
```diff
@@ -3,160 +3,201 @@
...
     "text-": {
       "secondary": {
-        "$value": "{color.base.blue-.600}"
+        "$value": "{color.base.blue-.600}",
+        "$description": "Text that is relevant but less important can be lower-contrast to suggest its’ reduced importance."
       },
       "reverse": {
-        "$value": "{color.white}"
+        "$value": "{color.white}",
+        "$description": "Use on “reversed” (dark) sections."
```
Comment (chris-at-jobber) [inline packages/design/src/tokens/semanticColor.tokens.json:134]:
```suggestion
        "$description": "Use for typographic content against reversed (inverted theme) surfaces."
```

File: packages/design/src/tokens/semanticColor.tokens.json [hunk 8]
```diff
@@ -3,160 +3,201 @@
...
       },
       "reverse-": {
         "secondary": {
-          "$value": "{color.base.blue-.300}"
+          "$value": "{color.base.blue-.300}",
+          "$description": "Secondary text for “reversed” sections."
```
Comment (chris-at-jobber) [inline packages/design/src/tokens/semanticColor.tokens.json:139]:
```suggestion
          "$description": "Secondary text for use against reversed (inverted theme) surfaces."
```

File: packages/design/src/tokens/semanticColor.tokens.json [hunk 9]
```diff
@@ -3,160 +3,201 @@
...
         }
       }
     },
     "icon": {
       "$value": "{color.base.blue-.800}"
     },
     "surface": {
-      "$value": "{color.white}"
+      "$value": "{color.white}",
+      "$description": "Most elements in Jobber have a white surface; if clickable, they have a light hover and active color."
```
Comment (chris-at-jobber) [inline packages/design/src/tokens/semanticColor.tokens.json:148]:
```suggestion
      "$description": "The primary color of the application surface (background). If interactive, surfaces have a hover and active color."
```

File: packages/design/src/tokens/semanticColor.tokens.json [hunk 10]
```diff
@@ -3,160 +3,201 @@
...
     "surface-": {
       "hover": {
-        "$value": "{color.base.taupe-.200}"
+        "$value": "{color.base.taupe-.200}",
+        "$description": "Hover state for surface."
       },
       "reverse": {
-        "$value": "{color.base.blue-.900}"
+        "$value": "{color.base.blue-.900}",
+        "$description": "When a strong constrast is needed with the rest of the application, a reversed surface can be used."
```
Comment (chris-at-jobber) [inline packages/design/src/tokens/semanticColor.tokens.json:157]:
```suggestion
        "$description": "Use a reverse surface to invert the theme when a strong contrast is needed with the rest of the application."
```

**Review by chris-at-jobber — CHANGES_REQUESTED** [review]
Looks great! Couple of suggestions to be less descriptive about "white" or "dark" surfaces as the lightness or darkness changes per theme/mode

**Review by scotttjob — COMMENTED** [review]
I ran through these quickly, did not run the code directly.

Can we run a quick diff on the output of the generated files before and after this change? There should be no change.

I normally wouldn't ask for a diff, but there's enough changes in here that a diff is a good sanity check.

**Comment by edison-cy-yang** [comment]
> I ran through these quickly, did not run the code directly.
> 
> Can we run a quick diff on the output of the generated files before and after this change? There should be no change.
> 
> I normally wouldn't ask for a diff, but there's enough changes in here that a diff is a good sanity check.

@scotttjob those generated css and js files are gitignored, by seeing the diff are you talking about temporarily adding those files to the index and running diff on them (then removing them from index)?

In this case would it make sense to include this step (in a shell script) as part of running `npm run bootstrap` just to confirm the diffs are what we expected?

**Comment by scotttjob** [comment]
> > I ran through these quickly, did not run the code directly.
> > Can we run a quick diff on the output of the generated files before and after this change? There should be no change.
> > I normally wouldn't ask for a diff, but there's enough changes in here that a diff is a good sanity check.
> 
> @scotttjob those generated css and js files are gitignored, by seeing the diff are you talking about temporarily adding those files to the index and running diff on them (then removing them from index)?
> 
> In this case would it make sense to include this step (in a shell script) as part of running `npm run bootstrap` just to confirm the diffs are what we expected?

I don't think we have to go that far, I was thinking just a manual diff. So if you swapped branches back to master, ran the build command, copied the files out, swapped branches, ran the command again, then manually compared.

So a one-off screenshot. :)

**Comment by edison-cy-yang** [comment]
> > > I ran through these quickly, did not run the code directly.
> > > Can we run a quick diff on the output of the generated files before and after this change? There should be no change.
> > > I normally wouldn't ask for a diff, but there's enough changes in here that a diff is a good sanity check.
> > 
> > 
> > @scotttjob those generated css and js files are gitignored, by seeing the diff are you talking about temporarily adding those files to the index and running diff on them (then removing them from index)?
> > In this case would it make sense to include this step (in a shell script) as part of running `npm run bootstrap` just to confirm the diffs are what we expected?
> 
> I don't think we have to go that far, I was thinking just a manual diff. So if you swapped branches back to master, ran the build command, copied the files out, swapped branches, ran the command again, then manually compared.
> 
> So a one-off screenshot. :)

Picked a few files

Screenshot 2024-11-07 at 10 40 59 AM
Screenshot 2024-11-07 at 10 41 16 AM
Screenshot 2024-11-07 at 10 41 47 AM

**Comment by scotttjob** [comment]
> > > > I ran through these quickly, did not run the code directly.
> > > > Can we run a quick diff on the output of the generated files before and after this change? There should be no change.
> > > > I normally wouldn't ask for a diff, but there's enough changes in here that a diff is a good sanity check.
> > > 
> > > 
> > > @scotttjob those generated css and js files are gitignored, by seeing the diff are you talking about temporarily adding those files to the index and running diff on them (then removing them from index)?
> > > In this case would it make sense to include this step (in a shell script) as part of running `npm run bootstrap` just to confirm the diffs are what we expected?
> > 
> > 
> > I don't think we have to go that far, I was thinking just a manual diff. So if you swapped branches back to master, ran the build command, copied the files out, swapped branches, ran the command again, then manually compared.
> > So a one-off screenshot. :)
> 
> Picked a few files
> 
> Screenshot 2024-11-07 at 10 40 59 AM Screenshot 2024-11-07 at 10 41 16 AM Screenshot 2024-11-07 at 10 41 47 AM

Amaaaaaaazing! That's exactly the kind the reassurance I was looking for. Great work Edi!

**Review by chris-at-jobber — APPROVED** [review]
Looks great! Thanks @edison-cy-yang

**Pull request merged**